src/DGtal/geometry/curves/estimation/LambdaMST3D.h

```diff
@@ -0,0 +1,181 @@
+/**
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ **/
+
+#pragma once
+
+/**
+ * @file LambdaMST3D.h
+ * @author Kacper Pluta (\c kacper.pluta@esiee.fr )
+ * Laboratoire d'Informatique Gaspard-Monge - LIGM, France
+ *
+ * @date 2014/10/06
+ *
+ * This file is part of the DGtal library.
+ */
+
+#if defined(LAMBDAMST3D_RECURSES)
+#error Recursive header files inclusion detected in LambdaMST3D.h
+#else // defined(LAMBDAMST3D_RECURSES)
+/** Prevents recursive inclusion of headers. */
+#define LAMBDAMST3D_RECURSES
+
+#if !defined LAMBDAMST3D_h
+/** Prevents repeated inclusion of headers. */
+#define LAMBDAMST3D_h
+
+#include <algorithm>
+#include <iterator>
+#include <cmath>
+#include <map>
+#include <DGtal/base/Common.h>
+#include <DGtal/helpers/StdDefs.h>
+#include "DGtal/kernel/CSpace.h"
+#include "DGtal/kernel/PointVector.h"
+#include "DGtal/geometry/curves/CIncrementalSegmentComputer.h"
+#include "DGtal/geometry/curves/estimation/FunctorsLambdaMST.h"
+
+namespace DGtal {
+  
+  template < typename TSpace, typename TSegmentation, typename Functor >
+  class LambdaMST3DEstimator
+  {
+  public: 
+    ///Checking concepts
+    BOOST_CONCEPT_ASSERT(( concepts::CSpace<TSpace> ));
+    BOOST_STATIC_ASSERT(( TSpace::dimension == 3 ));
+    BOOST_CONCEPT_ASSERT(( CIncrementalSegmentComputer<typename TSegmentation::SegmentComputer> ));
+    // ----------------------- Types ------------------------------
+  public:
+    typedef TSegmentation Segmentation;
+    typedef typename TSegmentation::SegmentComputer SegmentComputer;
+    typedef typename SegmentComputer::ConstIterator ConstIterator;
+    typedef typename Functor::Value Value;
+    typedef typename TSpace::RealVector RealVector;
+    typedef typename TSpace::Point Point;
+    
+    // ----------------------- Standard services ------------------------------
+  public:
+    LambdaMST3DEstimator();
+    
+    /**
+     * Initialisation.
+     * @param itb begin iterator
+     * @param ite end iterator
+     */
+    void init ( const ConstIterator& itb, const ConstIterator& ite );
+    
+    /**
+     * @param segmentComputer - DSS segmentation algorithm
+     */
+    void attach ( const TSegmentation & segmentComputer );
```

Comment:
Suggest "ConstAlias<Segmentation> segmentComputer"  if the lifetime of segmentComputer exceeds the lifetime of this method.